Minimum Viable Product

whitty12 opened the pull request:
Finished Project - Graphical Tweaks, updated original rspec tests so they should pass with the layers of authentication and the merged migrations

whitty12 committed:
Cart item broke again :(

whitty12 committed:
Fixed issue: 'cart id is nil' -> removed the before_action calling before the items creation

whitty12 committed:
Added navigation bar with functioning authorization using Bootstrap

whitty12 committed:
Version 1.0 - Finished Prototype

DianaLiao commented:
Fun idea! Before making my final review I'd love to see a few tweaks to put it over the edge:
- more validations in the User model
- error handling for said User validations
- add a default route for the app (I had to find it by typing in `/login` myself 😄 )
- bonus/stretch: add a seed file to include an admin user

whitty12 committed:
Added default root path to products page

whitty12 committed:
Added validation and testing for user, added extra testing for product and cart_item model specs

whitty12 committed:
Added uniqueness validation to email, fixed validation tests

whitty12 committed:
Fixed error message handling on login page - message will properly flash

whitty12 committed:
Adjusted account creation to redirect to login with user created notice

whitty12 commented:
Revisions made to the MVP branch:
- Removed stray header tags from the user form
- Added proper account login notifications including notifications of failure to login when the password or username is incorrect
- Added validations to the user model for username, first name, last name, zip, address, state, and email where email must be unique
- Added tests to spec/model for products, cart_item (changed from cart_items), and user
- User spec model now tests the required parameters that need to be validated (first name, last name, zip, etc.)
- Product spec now tests for required parameters (title, description, price)
- Cart_Item spec now tests for required cart_id and required product_id
- Adjust product#show to redirect to login instead of <p> tags saying to login to add an item to a cart
